### PR TITLE
MULE-10348: Processors in DefaultMessageProcessorChain are not comple…

### PR DESCRIPTION
**Pull request by mbuchwald**
MULE-10348: Processors in DefaultMessageProcessorChain are not completely initialized

### CODE REVIEW

File: core/src/main/java/org/mule/processor/chain/InterceptingChainLifecycleWrapperPathSkip.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.processor.chain;
+
+import org.mule.api.processor.MessageProcessor;
+import org.mule.api.processor.MessageProcessorChain;
+
+import java.util.List;
+
+public class InterceptingChainLifecycleWrapperPathSkip extends InterceptingChainLifecycleWrapper{
```
Comment (aiannucci) [inline core/src/main/java/org/mule/processor/chain/InterceptingChainLifecycleWrapperPathSkip.java]: Fix format

File: core/src/main/java/org/mule/processor/chain/PathSkipMessageProcessorChainBuilder.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.processor.chain;
+
+import org.mule.api.processor.MessageProcessorChain;
+
+public class PathSkipMessageProcessorChainBuilder extends DefaultMessageProcessorChainBuilder {
```
Comment (aiannucci) [inline core/src/main/java/org/mule/processor/chain/PathSkipMessageProcessorChainBuilder.java]: Fix format

File: core/src/main/java/org/mule/processor/chain/InterceptingChainLifecycleWrapperPathSkip.java (hunk 1 above)
Comment (mbuchwald) [inline core/src/main/java/org/mule/processor/chain/InterceptingChainLifecycleWrapperPathSkip.java]: fixed

File: core/src/main/java/org/mule/processor/chain/PathSkipMessageProcessorChainBuilder.java (hunk 2 above)
Comment (mbuchwald) [inline core/src/main/java/org/mule/processor/chain/PathSkipMessageProcessorChainBuilder.java]: fixed

File: core/src/main/java/org/mule/processor/chain/InterceptingChainLifecycleWrapperPathSkip.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.processor.chain;
+
+import org.mule.api.processor.MessageProcessor;
+import org.mule.api.processor.MessageProcessorChain;
+
+import java.util.List;
+
+public class InterceptingChainLifecycleWrapperPathSkip extends InterceptingChainLifecycleWrapper
```
Comment (dfeist) [inline core/src/main/java/org/mule/processor/chain/InterceptingChainLifecycleWrapperPathSkip.java]: Why do some uses of InterceptingChainLifecycleWrapper need to be skipped but others not?

Comment (mbuchwald) [inline core/src/main/java/org/mule/processor/chain/InterceptingChainLifecycleWrapperPathSkip.java]: We are now returning the chain when getting the MPs from Batch. Without skipping the chains, this added a level in the paths. Firstly I tried skipping the InterceptingChainLifecycleWrapper, but for Choice and ScatterGather this removed the last levels (the processors where wrapped with this chain).. So, I needed a type of Interceptor that could be skipped.

**Pull request merged**